src/cudadev/plugin-PixelTriplets/CAConstants.h

```diff
@@ -10,60 +10,73 @@
+  constexpr uint32_t maxCellsPerHit = 8 * 128;
+#else  // ONLY_PHICUT
+  constexpr uint32_t maxCellNeighbors = 36;
+  constexpr uint32_t maxCellTracks = 48;
 #ifdef GPU_SMALL_EVENTS
-  constexpr uint32_t maxNumberOfTuples() { return 3 * 1024; }
-#else
-  constexpr uint32_t maxNumberOfTuples() { return 24 * 1024; }
-#endif
-#else
-  constexpr uint32_t maxNumberOfTuples() { return 48 * 1024; }
-#endif
-  constexpr uint32_t maxNumberOfQuadruplets() { return maxNumberOfTuples(); }
-#ifndef ONLY_PHICUT
-#ifndef GPU_SMALL_EVENTS
-  constexpr uint32_t maxNumberOfDoublets() { return 512 * 1024; }
-  constexpr uint32_t maxCellsPerHit() { return 128; }
-#else
-  constexpr uint32_t maxNumberOfDoublets() { return 128 * 1024; }
-  constexpr uint32_t maxCellsPerHit() { return 128 / 2; }
-#endif
-#else
-  constexpr uint32_t maxNumberOfDoublets() { return 2 * 1024 * 1024; }
-  constexpr uint32_t maxCellsPerHit() { return 8 * 128; }
-#endif
-  constexpr uint32_t maxNumOfActiveDoublets() { return maxNumberOfDoublets() / 8; }
+  // kept for testing and debugging
+  constexpr uint32_t maxNumberOfTuples = 3 * 1024;
+  constexpr uint32_t maxNumberOfDoublets = 128 * 1024;
+  constexpr uint32_t maxCellsPerHit = 128 / 2;
+#else   // GPU_SMALL_EVENTS
+  // tested on MC events with 55-75 pileup events
+  constexpr uint32_t maxNumberOfTuples = 24 * 1024;
+  constexpr uint32_t maxNumberOfDoublets = 512 * 1024;
+  constexpr uint32_t maxCellsPerHit = 128;
+#endif  // GPU_SMALL_EVENTS
+#endif  // ONLY_PHICUT
+  constexpr uint32_t maxNumOfActiveDoublets = maxNumberOfDoublets / 8;
+  constexpr uint32_t maxNumberOfQuadruplets = maxNumberOfTuples;
 
-  constexpr uint32_t maxNumberOfLayerPairs() { return 20; }
-  constexpr uint32_t maxNumberOfLayers() { return 10; }
-  constexpr uint32_t maxTuples() { return maxNumberOfTuples(); }
+  constexpr uint32_t maxNumberOfLayerPairs = 20;
+  constexpr uint32_t maxNumberOfLayers = 10;
+  constexpr uint32_t maxTuples = maxNumberOfTuples;
+
+  // Modules constants
+  constexpr uint32_t max_ladder_bpx0 = 12;
+  constexpr uint32_t first_ladder_bpx0 = 0;
+  constexpr float module_length_bpx0 = 6.7f;
+  constexpr float module_tolerance_bpx0 = 0.4f;  // projection to cylinder is inaccurate on BPIX1
+  constexpr uint32_t max_ladder_bpx4 = 64;
+  constexpr uint32_t first_ladder_bpx4 = 84;
+  constexpr float radius_even_ladder = 15.815f;
+  constexpr float radius_odd_ladder = 16.146f;
+  constexpr float module_length_bpx4 = 6.7f;
+  constexpr float module_tolerance_bpx4 = 0.2f;
+  constexpr float barrel_z_length = 26.f;
+  constexpr float forward_z_begin = 32.f;
+
+  // Last indexes
+  constexpr uint32_t last_bpix1_detIndex = 96;
+  constexpr uint32_t last_barrel_detIndex = 1184;
 
   // types
-  using hindex_type = uint16_t;  // FIXME from siPixelRecHitsHeterogeneousProduct
-  using tindex_type = uint16_t;  //  for tuples
+  using hindex_type = uint32_t;  // FIXME from siPixelRecHitsHeterogeneousProduct
```

Comment:
here as well!

Reply:
all these changes and still in two places....

Reply:
if I change this I can run fine, but I still have a ~3.5% loss in performance

Reply:
so this is not the cause of the regression